### ISSUE TITLE
[AIRE] CI failure: NVPanoptix3Dv2 scripts' companion imports fail test_internal_imports in run 33495660045 (PR #123)

### ISSUE THREAD

**Issue by svc-bcs-agent** [NVIDIA/tao_pytorch_backend#124]
## Summary

Blossom-CI run [`33495660045`](https://github.com/NVIDIA-TAO/tao-pytorch/actions/runs/33495660045) for PR #123 (`[TAOVN-1251][Feature] Add NVPanoptix3Dv2 scripts, utilities, and experiment specs`) failed the repo-wide static import invariant:

```
=========================== short test summary info ============================
FAILED tests/test_imports.py::test_internal_imports - Failed:
Found import errors in 4 file(s):
=== 1 failed, 2428 passed, 153 skipped, 43557 warnings in 7505.30s (2:05:05) ===
```

The Jenkins build ended with `hudson.AbortException: script returned exit code 1`.

> **Note on detection:** the GitHub run `conclusion` for this run is `success`. Blossom-CI only mirrors the internal Jenkins log into the `Upload log` job, so the GitHub conclusion is not a reliable failure signal — this failure was detected by scanning the mirrored log content (job `99817130931`).

## Failing check

`tests/test_imports.py::test_internal_imports` — a static AST check that resolves every `nvidia_tao_pytorch.*` import to a file on disk **without importing it**.

<details>
<summary>Full failure output (16 unresolved imports across 4 files)</summary>

```
  nvidia_tao_pytorch/cv/nvpanoptix3d_v2/scripts/evaluate.py
    Line 18: Module 'nvidia_tao_pytorch.config.nvpanoptix3d_v2.default_config' not found on disk
    Line 19: Module 'nvidia_tao_pytorch.cv.nvpanoptix3d_v2.dataloader' not found on disk
    Line 20: Module 'nvidia_tao_pytorch.cv.nvpanoptix3d_v2.model.build_pl_model' not found on disk
    Line 31: Module 'nvidia_tao_pytorch.cv.nvpanoptix3d_v2.dataloader.panoptic.pl_data_module' not found on disk

  nvidia_tao_pytorch/cv/nvpanoptix3d_v2/scripts/export.py
    Line 24: Module 'nvidia_tao_pytorch.config.nvpanoptix3d_v2.default_config' not found on disk
    Line 25: Module 'nvidia_tao_pytorch.cv.nvpanoptix3d_v2.model.build_pl_model' not found on disk
    Line 26: Module 'nvidia_tao_pytorch.cv.nvpanoptix3d_v2.export.onnx_exporter' not found on disk
    Line 54: Module 'nvidia_tao_pytorch.cv.nvpanoptix3d_v2.dataloader.panoptic.pl_data_module' not found on disk

  nvidia_tao_pytorch/cv/nvpanoptix3d_v2/scripts/inference.py
    Line 20: Module 'nvidia_tao_pytorch.config.nvpanoptix3d_v2.default_config' not found on disk
    Line 21: Module 'nvidia_tao_pytorch.cv.nvpanoptix3d_v2.dataloader' not found on disk
    Line 22: Module 'nvidia_tao_pytorch.cv.nvpanoptix3d_v2.model.build_pl_model' not found on disk
    Line 33: Module 'nvidia_tao_pytorch.cv.nvpanoptix3d_v2.dataloader.panoptic.pl_data_module' not found on disk

  nvidia_tao_pytorch/cv/nvpanoptix3d_v2/scripts/train.py
    Line 24: Module 'nvidia_tao_pytorch.config.nvpanoptix3d_v2.default_config' not found on disk
    Line 25: Module 'nvidia_tao_pytorch.cv.nvpanoptix3d_v2.dataloader' not found on disk
    Line 26: Module 'nvidia_tao_pytorch.cv.nvpanoptix3d_v2.model.build_pl_model' not found on disk
    Line 53: Module 'nvidia_tao_pytorch.cv.nvpanoptix3d_v2.dataloader.panoptic.pl_data_module' not found on disk
```

</details>

## Root cause

PR #123 is the **scripts slice** of a deliberate multi-PR split of TAOVN-1251. It adds only
`nvidia_tao_pytorch/cv/nvpanoptix3d_v2/{scripts,utils,entrypoint,experiment_specs}`, but its four
entrypoint scripts import four module trees that are delivered by *companion* patches in the same
series and are therefore not on disk yet:

| Module | Delivered by |
|---|---|
| `nvidia_tao_pytorch.config.nvpanoptix3d_v2.default_config` | config patch |
| `nvidia_tao_pytorch.cv.nvpanoptix3d_v2.dataloader` (+ `.panoptic.pl_data_module`) | dataloader patch |
| `nvidia_tao_pytorch.cv.nvpanoptix3d_v2.model.build_pl_model` | model patch |
| `nvidia_tao_pytorch.cv.nvpanoptix3d_v2.export.onnx_exporter` | export patch |

`tests/test_imports.py` enforces this invariant **repo-wide on every PR**, so any single slice of the
split fails until all slices land. This is not a missing `git add` — the paths are not gitignored,
they simply belong to other PRs in the series.

This is the same failure class already handled for the sibling slice in PR #122 (utils) — see issue #121.

## Fix

Wrap the not-yet-landed companion imports in `try/except ImportError` and fall back to placeholders
that raise a **descriptive** `ImportError` the moment the symbol is actually used.

`tests/test_imports.py` has an explicit, designed escape hatch for exactly this
(`_collect_optional_import_nodes` / `_handler_catches_import_error`): imports inside a `try:` whose
handler catches `ImportError` are skipped by the static check. The fix uses that supported mechanism
rather than weakening the invariant.

One refinement over the PR #122 pattern: PR #123's scripts import **constants** as well as callables
(`PANOPTIC`, `INPUT_NAME`) and compare them with `==`/`!=`. A call-only placeholder would compare
unequal *silently* (`train.py`) or surface a misleading `ValueError` (`export.py`, where `PANOPTIC` is
the first placeholder touched). The placeholder therefore raises on `__call__`, `__eq__` and `__ne__`,
while keeping `__repr__` safe so tracebacks and debuggers render without a secondary failure.

Once the companion packages land, the `try` branch simply succeeds and behavior is unchanged; the
guards are self-contained per file and trivially revertible.

## Verification

Reproduced and verified locally against the exact `_check_file` / `_find_python_files` code path the
CI job runs:

- before the fix: `scanned 1661 files; 4 file(s) with import errors` (exit 1) — byte-identical to CI
- after the fix: `scanned 1661 files; 0 file(s) with import errors` (exit 0)
- all four changed files pass `python -m py_compile`

## Links

- Failing run: https://github.com/NVIDIA-TAO/tao-pytorch/actions/runs/33495660045
- `Upload log` job: https://github.com/NVIDIA-TAO/tao-pytorch/actions/runs/33495660045/job/99817130931
- PR under test: #123 (`feature/nvp2-script` @ `5268ba22a3a92febdb3641ae5e79566294426235`)
- Sibling slice: #122 / #121

## Suggested labels

`bug`, `ci`, `AIRE`

---
*Filed automatically by AIRE (AI Reliability Engineer) from Blossom-CI run `33495660045`.*

**Comment by svc-bcs-agent** [comment]
Fix PR opened: #125 (`svc-bcs-agent:fix/tao-pytorch-5268ba2` → `feature/nvp2-script`).

Verified locally against the exact `tests/test_imports.py` code path: 4 files with import errors before, 0 after. Blossom-CI validation is running on the fix PR.

**Comment by svc-bcs-agent** [comment]
### 🤖 AIRE self-heal update — verification build of PR #125

The verification build for the fix in PR #125 ([run 33499373447](https://github.com/NVIDIA-TAO/tao-pytorch/actions/runs/33499373447)) did **not** reproduce the original `test_internal_imports` failure — that fix appears to be working. However, the build still went red for an **unrelated reason**.

**New finding — `infra_flake` (not a regression from PR #123 or PR #125):**

- Jenkins terminated with `hudson.AbortException: script returned exit code 139` (`139 = 128 + 11` → **SIGSEGV**).
- Crash line: `Segmentation fault (core dumped) ... pytest -v --testmon tests/ --cov=nvidia_tao_pytorch`
- Crashed during `tests/cv_unit_test/backbone_v2/test_backbone.py::test_basic_usage[freeze_at_1-ac_off-convnext_tiny]`, at ~3% suite progress.
- Test tally before the crash: **97 passed, 2 skipped, 0 failed, 0 errors** — there was no Python-level failure; the interpreter died natively with no traceback.

**Why this is not caused by the code under test:**

1. The **same** test passed in `0.99s` in the immediately-preceding build ([run 33495660045](https://github.com/NVIDIA-TAO/tao-pytorch/actions/runs/33495660045)) on the same branch.
2. PR #123 changes 22 files, **all** under `nvidia_tao_pytorch/cv/nvpanoptix3d_v2/`; PR #125 changes 4 files in that same directory.
3. Neither PR touches `cv/backbone_v2/`, `core/`, `conftest.py`, `pytest.ini`, or any dependency specification — so `convnext_tiny` backbone instantiation is not reachable from the changed code.
4. Environment was identical across both builds (driver `580.86`, CUDA `13.2`); only the ephemeral CI node differed (`…-143-…` vs `…-144-…`).

**Remedy:** re-trigger the pipeline. Note that `pytest-rerunfailures` and `pytest-flakefinder` are both installed, but **neither can recover a SIGSEGV** — an interpreter crash kills the whole pytest process, so in-process retry plugins never get to run. Only a pipeline-level re-run helps.

AIRE is re-triggering `/build` on PR #125 now and will report the result.

*If this segfault recurs across multiple independent nodes, it should be treated as a genuine defect in the `backbone_v2` convnext path (or its native deps) and triaged separately from this PR.*

**Comment by svc-bcs-agent** [comment]
✅ **Verified green.** Blossom-CI [run 33522000919](https://github.com/NVIDIA-TAO/tao-pytorch/actions/runs/33522000919) on fix PR #125:

```
tests/test_imports.py::test_internal_imports PASSED
======== 2429 passed, 153 skipped, 43563 warnings in 9152.27s (2:32:32) ========
```

Zero failures; SonarScanner `EXECUTION SUCCESS`. Compare the original failing run [33495660045](https://github.com/NVIDIA-TAO/tao-pytorch/actions/runs/33495660045): `1 failed, 2428 passed`.

An intermediate build ([33499373447](https://github.com/NVIDIA-TAO/tao-pytorch/actions/runs/33499373447)) aborted with `exit code 139` — a `Segmentation fault` in `backbone_v2/test_backbone.py::test_basic_usage[freeze_at_1-ac_off-convnext_tiny]` that killed pytest at 3%. That is the pre-existing flaky infra crash tracked in #104: the same test passed in the original 2h05m run with zero segfaults, and it does not touch any code this fix changes. It did not reproduce on retrigger.

Awaiting review/merge of #125 into `feature/nvp2-script`.

**Comment by svc-bcs-agent** [comment]
### ✅ AIRE self-heal result — build is GREEN

Follow-up to the previous note: the re-triggered build has completed and **passed**.

**[Run 33522000919](https://github.com/NVIDIA-TAO/tao-pytorch/actions/runs/33522000919)** (correlated to PR #125): **2429 passed, 153 skipped, 0 failed** — 100% of the suite, in 2:32:32.

Two confirmations:

1. `tests/cv_unit_test/backbone_v2/test_backbone.py::test_basic_usage[freeze_at_1-ac_off-convnext_tiny]` — the test that segfaulted in run 33499373447 — **passed in `1.00s`**. This confirms the `infra_flake` classification: the crash did not reproduce, and no code change was needed.
2. `tests/test_imports.py::test_internal_imports` — the original bug this issue tracks — **passed** (19.85s). The fix in PR #125 holds.

No source changes were made in this iteration; the remedy was a pipeline re-trigger, which is the only valid remedy for a SIGSEGV (in-process retry plugins cannot recover an interpreter crash).

PR #125 is green and ready for review/merge. Closing the loop here — this issue can be closed once PR #125 merges.
